store/tikv/region_cache.go

```diff
@@ -462,42 +463,92 @@ func (c *RegionCache) loadRegionByID(bo *Backoffer, regionID uint64) (*Region, e
 			return nil, errors.New("receive Region with no peer")
 		}
 		region := &Region{
-			meta: meta,
-			peer: meta.Peers[0],
+			meta:      meta,
+			_workPeer: unsafe.Pointer(meta.Peers[0]),
 		}
 		if leader != nil {
-			region.SwitchPeer(leader.GetStoreId())
+			region.SwitchWorkPeer(leader.GetStoreId())
 		}
 		return region, nil
 	}
 }
 
-// GetStoreAddr returns a tikv server's address by its storeID. It checks cache
-// first, sends request to pd server when necessary.
-func (c *RegionCache) GetStoreAddr(bo *Backoffer, id uint64) (string, error) {
-	c.storeMu.RLock()
-	if store, ok := c.storeMu.stores[id]; ok {
+func (c *RegionCache) getRegionCacheItemWithRLock(regionID RegionVerID) (cacheItem *CachedRegion) {
+	c.mu.RLock()
+	cacheItem = c.mu.regions[regionID]
+	c.mu.RUnlock()
+	return
+}
+
+func (c *RegionCache) ensureRegionWorkPeer(region *Region, ts int64) (workStore *Store, workPeer *metapb.Peer) {
+retry:
+	regionPeer := region.WorkPeer()
+
+	var (
+		cachedStore *Store
+		exists      bool
+	)
+	if regionPeer != nil {
+		c.storeMu.RLock()
```

Comment:
If we put store access pointer in `CachedRegion`, then we don't need to lock the storeMu and read from stores map.